src/runtime/utils.ts

```diff
@@ -100,3 +102,35 @@ export function trapUnhandledNodeErrors() {
     );
   }
 }
+
+const joinIfArray = (value: string | string[]) =>
+  Array.isArray(value) ? value.join(",") : value;
+
+export function normalizeOutgoingHeaders(
+  headers: Awaited<ReturnType<NitroApp["localCall"]>>["headers"] | Headers
+) {
+  const outgoingHeaders = new Headers();
+  const iterableHeaders =
+    headers instanceof Headers ? headers : Object.entries(headers);
+  for (const [name, header] of iterableHeaders) {
+    if (name === "set-cookie") {
+      for (const cookie of splitCookiesString(joinIfArray(header))) {
+        outgoingHeaders.append("set-cookie", cookie);
+      }
+    } else {
+      outgoingHeaders.set(name, joinIfArray(header));
+    }
+  }
+  return outgoingHeaders;
+}
+
+export function withNormalizedHeaders(fetch: typeof globalThis.fetch) {
```

Comment:
This as a generic utility might be confusing to be seen (and named like this) here. Do you mind to move it inline to `app.ts`? 🙏🏼 (also we might end up moving fix finally to the unenv)